spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -525,12 +531,6 @@ else if (listener instanceof MessageListener) {
 				this.batchErrorHandler = new BatchLoggingErrorHandler();
 			}
 			Assert.state(!this.isBatchListener || !this.isRecordAck, "Cannot use AckMode.RECORD with a batch listener");
-			if (this.transactionManager != null) {
-				this.transactionTemplate = new TransactionTemplate(this.transactionManager);
-			}
-			else {
-				this.transactionTemplate = null;
-			}
```

Comment:
FYI, this move would not be needed for production since `onPartitionsAssigned()` won't be called until we `poll()`.

It is, needed, however, for mocked tests since it is called directly downstream of `subscribeOrAssignTopics()`.